### DISCUSSION TITLE
Syncing messages/calls between two phones. Migrating from two phones into one.

### DISCUSSION THREAD

**Discussion by przemoc** [tmo1/sms-ie#173] [General]
Imagine someone was switching one SIM card between two phones (phone A and phone B), so SMS/MMS  messages and calls are scattered between them.

Are following scenarios supported:

1. Importing messages/calls from phone A onto phone B but retaining messages/calls that were already on phone B?
2. Importing messages/calls from phone A and importing messages/calls from phone B onto new phone C (no prior messages/calls)?

I did some experiment (actually by accident, forgetting to remove call logs before new fresh import, as `sms-ie` doesn't have button for removing call logs, so it needs to be done in dedicated app handling call history), and it looks that old call entries got doubled. I haven't tried it with messages. So it looks there may be some level of support for 1 or 2, at least for calls, but it feels like it may lack support of preventing adding (duplicating) messages/calls that are already in the device.

I may create dedicated issues later for all of that but wanted to start with a discussion.

**Comment by tmo1** [comment]
> Are following scenarios supported:

I haven't tried, but I don't see why they shouldn't work.

> it looks that old call entries got doubled.

> it may lack support of preventing adding (duplicating) messages/calls that are already in the device.

[Some support for message deduplication upon import is available](https://github.com/tmo1/sms-ie#deduplication). There is currently no support for call log deduplication.

**Reply by przemoc** [reply]
Call log deduplication is available starting from:
https://github.com/tmo1/sms-ie/releases/tag/v2.4.0
Thanks for that. I have not tested it yet.
